### PR TITLE
New Projectile Effect:Temperature Change

### PR DESCRIPTION
**Pull request by Boaz1111**
## About the PR

new component called TemperatureChangeOnCollideComponent that changes the temperature of whatever entity it hits. 
## Why / Balance

Requested for #30974 and #30784
## Technical details

Basically just calls _temperature.ChangeHeat(...) 
Component has two datafields, one that defines the amount of heat it adds or removes in joules, and one that defines if it does or doesn't ignore things such as winter coats.

## Media

https://github.com/user-attachments/assets/1037c211-c4b0-48d6-bc36-f8e514b8d8a0

https://github.com/user-attachments/assets/26ed1ab9-1b10-44d7-8358-b04344515856

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

shouldn't be any
**Changelog**

not player facing

### CODE REVIEW

File: Content.Server/Temperature/Systems/TemperatureSystem.cs [hunk 1]
```diff
@@ -21,6 +23,7 @@ public sealed class TemperatureSystem : EntitySystem
     [Dependency] private readonly AtmosphereSystem _atmosphere = default!;
     [Dependency] private readonly DamageableSystem _damageable = default!;
     [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly TemperatureSystem _temperatureSystem = default!;
```
Comment (TheShuEd) [inline Content.Server/Temperature/Systems/TemperatureSystem.cs:26]:
```suggestion
    [Dependency] private readonly TemperatureSystem _temperature = default!;
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 2]
```diff
@@ -345,16 +345,16 @@
...
     #   soundHit:  Waiting on serv3
     impactEffect: BulletImpactEffectDisabler
     damage:
       types:
-        Cold: 10
+        Cold: 0
   - type: TimedDespawn
     lifetime: 3
+  - type: ChangeTemperatureOnCollide
+    heat: -50000
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:357]: Wtf lol
50000?
Its normal?

Comment (Boaz1111) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:357]: It's in Joules, and yes, this cools a human target down by approximately 20 degrees celcius per hit.

**Comment by Boaz1111** [comment]
@TheShuEd your requested changes seem to have broken it, do I revert them?

**Comment by TheShuEd** [comment]
No, you just need to change the name of the variable throughout the file where it is used

**Comment by Boaz1111** [comment]
ah. Well, currently don't have access to my computer, so I'll have to do that later, seeing as webedit's not a good idea

**Pull request merged**